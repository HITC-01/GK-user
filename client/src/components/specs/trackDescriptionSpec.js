/* eslint-env jest */
/* eslint react/jsx-filename-extension: 0 */

import React from 'react';
import renderer from 'react-test-renderer';
import TrackDescription from '../trackDescription.jsx';
import HashTag from '../hashtag.jsx';
import { shallow } from '../../enzyme';

describe('<TrackDescription />', () => {
  test('Should render correctly', () => {
    const component = renderer.create(<TrackDescription />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should not render the HashTag Component if there are no hashes ', () => {
    const props = {
      description: 'Testing for the track description',
      hashtags: '',
      id: 80,
      userId: 90,
    };
    const component = shallow(<TrackDescription songData={props} />);

    expect(component.find(HashTag).length).toBe(0);
  });

  test('Should render the HashTag Component if there are hashes ', () => {
    const propsWithHash = {
      description: 'Testing for the track description',
      hashtags: 'Helllo',
      id: 80,
      userId: 90,
    };
    const component = shallow(<TrackDescription songData={propsWithHash} />);
    expect(component.find(HashTag).length).toBe(1);
  });

  test('Should change class of show / hide based on state change', () => {
    const component = shallow(<TrackDescription />);
    expect(component.find('.utd-show').length).toBe(1);
    component.setState({ isToggleOn: false });
    expect(component.find('.utd-show').length).toBe(0);
  });
});
