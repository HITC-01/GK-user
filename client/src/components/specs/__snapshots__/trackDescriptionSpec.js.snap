// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackDescription /> Should render correctly 1`] = `
<div
  className="utd-wrapper"
>
  <div
    className="utd-panel utd-show "
  >
    <p
      className="utd"
    >
      
    </p>
    <div
      className="utd-hash"
    >
      
    </div>
  </div>
  <div
    className="up-fade-show"
  />
  <br />
  <div
    className="utd-button-div"
  >
    <button
      className="utd-toggle"
      onClick={[Function]}
      type="button"
    >
      --- Show More v ---
    </button>
  </div>
</div>
`;
